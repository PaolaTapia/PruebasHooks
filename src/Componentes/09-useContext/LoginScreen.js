import React, { useContext, useState } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {

  const { user, setUser } = useContext(UserContext);

  return (
    <div>
      <h1>Login Screen</h1>
      <hr />
      <pre>
          {JSON.stringify(user,null,3)}
      </pre>

      <button
        className="btn btn-primary"
        onClick={() =>
          setUser({
            id: 123,
            user: "Paola",
          })
        }
      >SET USER</button>
    </div>
  );
};
