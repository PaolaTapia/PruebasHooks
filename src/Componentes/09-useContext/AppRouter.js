import React from 'react'; 
import {
     BrowserRouter as Router,
     Switch,
     Route,
     Redirect,
     Link
   } from "react-router-dom";
import { NavBar } from '../07-tarea-memo/NavBar';
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';


export const AppRouter = () => {
     return (
          <Router>
          <div>
            
          <NavBar />
          <div className="container">
            <Switch>
              <Route exact={true} path="/" component={HomeScreen}/>
              <Route exact path="/about" component={AboutScreen}/>
              <Route exact path="/login"  component={LoginScreen}/>            
              <Redirect to="/"/>            
           
            </Switch>
            </div>
          </div>
        </Router>
     );
}

