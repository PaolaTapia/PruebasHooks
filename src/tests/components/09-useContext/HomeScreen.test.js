import { mount } from 'enzyme';
import React from 'react';
import { HomeScreen } from '../../../Componentes/09-useContext/HomeScreen';
import { UserContext } from '../../../Componentes/09-useContext/UserContext';

describe('Pruebas en HomeScreen', () => {
     const user = {
          name: 'Paola',
          email: 'paola@gmail.com'
     }
     const wrapper= mount(
     <UserContext.Provider value={{
          user
     }}>
               <HomeScreen />
     </UserContext.Provider>
     ); 

     test('Debe de mostrarse correctamente ', () => {

          expect(wrapper).toMatchSnapshot(); 
     })
     
})
