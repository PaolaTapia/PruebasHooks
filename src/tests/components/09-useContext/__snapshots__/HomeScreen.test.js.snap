// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en HomeScreen Debe de mostrarse correctamente  1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
   "name": "Paola",
   "email": "paola@gmail.com"
}
  </pre>
</div>
`;
